atom/browser/api/atom_api_app.cc

```diff
@@ -912,6 +923,47 @@ void App::GetFileIcon(const base::FilePath& path,
   }
 }
 
+std::vector<mate::Dictionary> App::GetAppMemoryInfo(v8::Isolate* isolate) {
+  AppIdProcessIterator process_iterator;
+  auto processEntry = process_iterator.NextProcessEntry();
+  std::vector<mate::Dictionary> result;
+
+  while (processEntry != nullptr) {
+    int64_t pid = processEntry->pid();
+    auto process = base::Process::OpenWithExtraPrivileges(pid);
+
+#if defined(OS_MACOSX)
+    std::unique_ptr<base::ProcessMetrics> metrics(
+      base::ProcessMetrics::CreateProcessMetrics(
+        process.Handle(), content::BrowserChildProcessHost::GetPortProvider()));
+#else
+    std::unique_ptr<base::ProcessMetrics> metrics(
+      base::ProcessMetrics::CreateProcessMetrics(process.Handle()));
+#endif
+
+    mate::Dictionary pidDict = mate::Dictionary::CreateEmpty(isolate);
```

Comment:
If you're bored, you could change `processEntry`, `pidDict`, and `memoryDict` to use snake case. (Sorry. 🙃)